packages/headless/src/database/mongodb/super-admin/super-admin.model.ts

```diff
@@ -0,0 +1,58 @@
+import { SuperAdmin } from 'src/entity/super-admin';
+import { Schema, model } from 'mongoose';
+import { randomUUID } from 'crypto';
+
+const SuperAdminSchema = new Schema<SuperAdmin>(
+  {
+    id: {
+      type: String,
+      default: () => randomUUID(),
+      unique: true,
+    },
+    firstName: {
+      type: String,
+      trim: true,
+      required: true
+    },
+    lastName: {
+      type: String,
+      trim: true,
+      required: true
+    },
+    role: {
+      type: String,
+      trim: true,
+      required: true
+    },
+    email: {
+      type: String,
+      index: true,
+      unique: true,
+      required: true
+    },
+    countryCode: {
+      type: String,
+    },
+    phone: {
+      type: String,
+      index: true,
+      unique: true
```

Comment:
as phone number is optipnal, if two admin has phone: null, then unique constraint will fail. so we can rely on service layer validation. Though there's an alternate solution. partial indexing can be done, but that might not work in other dbms. so remove unique index.

Reply:
Thanks vai for pointing out this point.